sway-parse/src/token.rs

```diff
@@ -120,17 +120,34 @@ pub fn lex_commented(
         path,
         stream,
     };
+    let mut gather_module_docs = false;
+    let mut file_start_offset: usize = 0;
 
     let mut parent_token_trees = Vec::new();
     let mut token_trees = Vec::new();
     while let Some((mut index, mut character)) = l.stream.next() {
         if character.is_whitespace() {
+            // if the beginning of a file starts with whitespace
+            // we must keep track to ensure that the module level docs
+            // will get inserted into the tree correctly
+            if index - file_start_offset == 0 {
+                file_start_offset += character.len_utf8();
```

Comment:
hey @mitchmindtree just wanted to make sure this was the correct approach to managing the offset of whitespace at the beginning of a file

`.len_utf8()` returns the byte span in `usize`

Reply:
Hmm I'm not sure I fully understand why we need to keep track of the whitespace at the beginning. How does the whitespace affect whether or not the module level docs are inserted into the tree correctly?

Reply:
If in doubt, maybe we can just add more tests with weird whitespace gaps, e.g.

This one should probably be valid
```rust
// <comment or whitespace here>
//! some docs
```

Should this fail or merge with the top? Not sure what Rust does here:
```rust
//! some docs

//! more docs
```
Although it should *probably* merge, particularly if gapped by a normal comment, e.g. I guess the following should probably pass?
```rust
//! some docs
// //! some more commented out docs
//! some more docs

Reply:
Well the only way I could think of how to track module doc comments was to use the index. If the index - offset of whitespace is 0 then the bool `gather_module_docs` will be true so it will start to collect them as `DocStyle::Inner` `DocComment`s until the next char it finds is not a comment. It will continue to process through other normal comments just in case. So 
```rust
//! hello...
// hello
/* hi again */
//! world!
```
will produce two inner doc comments even though there are other comments in between.

Reply:
It will work in all cases you provided, I believe.